src/java/org/apache/cassandra/index/sai/disk/v3/BlockTerms.java

```diff
@@ -0,0 +1,334 @@
+        return Math.max(0, mismatch);
+    }
+
+    public static ByteComparable fixedLength(BytesRef bytes)
+    {
+        return ByteComparable.fixedLength(bytes.bytes, bytes.offset, bytes.length);
+    }
+
+    /**
+     * Appends unsigned byte 0 to the array to order correctly.
+     * @param orig Original byte array to operate on
+     * @return A new byte array that sorts one unit ahead of the given byte array
+     */
+    public static byte[] nudge(byte[] orig)
+    {
+        byte[] bytes = Arrays.copyOf(orig, orig.length + 1);
+        bytes[orig.length] = (int) 0; // unsigned requires cast
+        return bytes;
+    }
+
+    /**
+     * @param orig Prefix bytes
+     * @param maxLength The correct ordering the absolute max bytes length of all compared bytes is specified.
+     * @return Byte array that is the max of the given prefix.
+     */
+    public static byte[] prefixMaxTerm(byte[] orig, int maxLength)
+    {
+        assert orig.length > 0 : "Input byte array must be not empty";
+
+        byte[] result = Arrays.copyOf(orig, maxLength);
+        Arrays.fill(result, orig.length, result.length, (byte) 255);
+
+        return result;
+    }
+
+    /**
+     * @param orig Byte array to opreate on
+     * @param maxLength The correct ordering the absolute max bytes length of all compared bytes is specified.
+     * @return A byte array that sorts 1 byte unit increment lower
+     */
+    public static byte[] nudgeReverse(byte[] orig, int maxLength)
+    {
+        assert orig.length > 0 : "Input byte array must be not empty";
+        int lastIdx = orig.length - 1;
+
+        if (UnsignedBytes.toInt(orig[lastIdx]) == 0)
+            return Arrays.copyOf(orig, lastIdx);
+        else
+        {
+            // bytes must be max length for correct ordering
+            byte[] result = Arrays.copyOf(orig, maxLength);
+            result[lastIdx] = unsignedDecrement(orig[lastIdx]);
+            Arrays.fill(result, orig.length, result.length, (byte) 255);
+            return result;
+        }
+    }
+
+    public static byte unsignedDecrement(byte b)
+    {
+        int i = UnsignedBytes.toInt(b);
+        int i2 = i - 1; // unsigned decrement
+        return (byte) i2;
+    }
+
+    public static class LongArrayImpl implements LongArray
+    {
+        final LongArrayList list;
+
+        public LongArrayImpl(LongArrayList list)
+        {
+            this.list = list;
+        }
+
+        @Override
+        public long get(long idx)
+        {
+            return list.getLong((int)idx);
+        }
+
+        @Override
+        public long length()
+        {
+            return list.size();
+        }
+
+        @Override
+        public long findTokenRowID(long targetToken)
+        {
+            throw new UnsupportedOperationException();
```

Comment:
It's unused so no point in implementing it.

Reply:
This would be good to document so that the next reader knows it simply has no current uses and not, for example, that the operation is somehow incompatible with long arrays.